features/step_definitions/events.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+const Assert = require('chai').assert;
+const request = require('../support/request');
+
+module.exports = function server() {
+    // eslint-disable-next-line new-cap
+    this.Before({
+        tags: ['@events']
+    }, () => {
+        this.repoOrg = 'screwdriver-cd-test';
+        this.repoName = 'functional-events';
+        // Reset shared information
+        this.pipelineId = null;
+        this.eventId = null;
+    });
+
+    this.Given(/^an existing pipeline with the workflow:$/,
+        { timeout: 60 * 1000 }, table =>
+        this.getJwt(this.accessKey)
+        .then((response) => {
+            this.jwt = response.body.token;
+
+            return request({
+                uri: `${this.instance}/${this.namespace}/pipelines`,
+                method: 'POST',
+                auth: {
+                    bearer: this.jwt
+                },
+                body: {
+                    checkoutUrl: `git@github.com:${this.repoOrg}/${this.repoName}.git#master`
+                },
+                json: true
+            });
+        })
+        .then((response) => {
+            if (!this.pipelineId) {
+                if (response.statusCode === 201) {
+                    this.pipelineId = response.body.id;
+                } else if (response.statusCode === 409) {
+                    const str = response.body.message;
+                    const id = str.split(': ')[1];
+
+                    this.pipelineId = id;
+                }
+            }
+
+            Assert.oneOf(response.statusCode, [409, 201]);
+
+            return table;
+        })
+    );
+
+    this.Given(/^"calvin" has admin permission to the pipeline$/, () => null);
+
+    this.Then(/^an event is created$/, { timeout: 60 * 1000 }, () =>
+        request({
+            uri: `${this.instance}/${this.namespace}/pipelines/${this.pipelineId}/events`,
+            method: 'GET',
+            json: true
+        }).then(response => Assert.equal(response.body[0].id, this.eventId))
+    );
+
+    this.Then(/^the "main" build succeeds$/, { timeout: 60 * 1000 }, () =>
+        this.waitForBuild(this.buildId).then((resp) => {
```

Comment:
I'm actually using this one
https://github.com/screwdriver-cd/screwdriver/blob/master/features/support/before-hook.js#L52